app.py

```diff
@@ -1,67 +1,129 @@
 import datetime
 
 import requests
-from flask import Flask, render_template
+from flask import Flask, render_template, request
 from flask_caching import Cache
 from htmlmin.minify import html_minify
+from constants import CACHE_1_ZIP, CACHE_2_ZIP, CACHE_3_ZIP
+import re
+
+cache = Cache(config=
+    {
+        'CACHE_TYPE': 'simple',
+        'CACHE_DEFAULT_TIMEOUT': 1800
+    })
 
 app = Flask(__name__)
 
-cache = Cache(config={'CACHE_TYPE': 'simple'})
 cache.init_app(app)
 
-
-@app.route('/')
-@app.route('/index')
-@cache.cached(timeout=300)  # 1800 seconds == 30 minutes
-def index():
-
-    timestamp = datetime.datetime.now()
-
-    # // TODO -- MAKE THIS ACCEPT ANY LAT LONG AS A INPUT
-    # latitude and Longitude and  of the area
-    # NYC KNYC central park weather station is Lat: 40.78°N Lon: 73.97°W
-    # long is not negative here because "TOP" in the request accounts for this.
-
-    # Fetching weather data
-    # Hardcoding to the correct weather station
-    weather_data = requests.get(
-        f'https://api.weather.gov/gridpoints/OKX/33,35/forecast'
+def getContext(lat, lng, default_forecast, timestamp):
+    area_data = requests.get(
+        f'https://api.weather.gov/points/{lat},{lng}'
     ).json()
+    
+    cwa, gridX, gridY = area_data['properties']['cwa'], area_data['properties']['gridX'], area_data['properties']['gridY']
 
-    # Period wise data
-    periodic_data = weather_data['properties']['periods']
-
-    latitude, longitude = weather_data['geometry']['geometries'][0]['coordinates'][
-        0], weather_data['geometry']['geometries'][0]['coordinates'][1]
-
-    # Fetching area data
-    area_data = requests.get(
-        f'https://api.weather.gov/points/{longitude},{latitude}'
+    weather_data = requests.get(
+        f'https://api.weather.gov/gridpoints/{cwa}/{gridX},{gridY}/forecast'
     ).json()
 
-    print(area_data)
+    try:
+        periodic_data = weather_data['properties']['periods']
+    except:
+        context = {
+            'error': True,
+            'title': weather_data['title'],
+            'description': weather_data['detail']
+        }
+        return render_template('index.html', **context)
 
-    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state']
+    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state'] +['',' (Default)'][default_forecast]
 
     context = {
         'timestamp': timestamp,
         'periodic_data': periodic_data,
         'areaDescription': areaDescription
     }
 
-    return render_template('index.html', **context)
+    return context
+
+@app.route('/')
+@app.route('/index')
+@app.route('/<zip_code>')
+def index(zip_code=None):
+    
+    timestamp = datetime.datetime.now()
+
+    if zip_code != None and re.search(r'.*(\d{5}?)$', zip_code):
```

Comment:
Please remove all new cache logic, let's tackle that in a separate pull request!